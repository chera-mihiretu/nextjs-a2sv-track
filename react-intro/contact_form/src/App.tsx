import ContactFrom from "./components/ContactFrom";

function App() {
 

  return (
    <>
    <div className="mx-auto" style={{ width: "100vw" }}>
      <ContactFrom />
    </div>
    </>
  );
}

export default App
