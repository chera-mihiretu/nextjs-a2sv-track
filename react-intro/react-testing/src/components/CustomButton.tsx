
interface Props {
  onClick : () => void
}

const CustomButton = ({onClick} : Props) => {
  return (
    <button onClick={onClick}>Click me</button>
  )
}


export default CustomButton
