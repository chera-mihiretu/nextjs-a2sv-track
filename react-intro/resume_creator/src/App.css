#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}



form {
  border-radius: 10px;
  box-shadow: 0 0 2px rgb(161, 161, 161);

  display: flex;
  flex-direction: column;
  padding: 20px 40px 20px 40px;
}

input, textarea {
  padding: 12px;
  margin: 10px;
  border: 2px solid #ccc;
  outline: none;
  border-radius: 8px;  transition: border-color 0.3s, box-shadow 0.3s;
  
}

input:hover, textarea:hover  {
  box-shadow: 0 0 2px rgb(0, 136, 255);
}

/*next button */
.next {
  color: white;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  background-color: #0088ff;


}



/*button */
.btn {
  border: #0088ff 1px solid;

  margin: 10px 0 10px 0;
  transition: .3s ease;

}

.btn:hover {
  box-shadow: 0 0 0 5px rgba(0, 136, 255, .3);
  outline: none;
}

.btn:active {
  box-shadow: 0 0 0 3px rgba(0, 136, 255, .8);
  background-color: #0088ff;
  color: white;
  outline: none;
}


/*previous button*/

.previous {
  color: #0088ff;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  background-color: transparent;
}




.btn-container {
  display: flex;
  justify-content: center;
  width: 100%;
}

.btn-container > * {
  flex: 1;
}

