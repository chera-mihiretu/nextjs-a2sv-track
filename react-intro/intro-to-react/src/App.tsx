
import './App.css';
import Board from './components/Board';
import Game from './Game';


function App() {
  return (
    <div className="App" >

      <h1>This is personal</h1>
      <Game/>
    </div>
  );
}

export default App;
