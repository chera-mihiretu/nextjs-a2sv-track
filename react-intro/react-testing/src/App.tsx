import './App.css'
import { useState } from 'react';

function App() {
  // You can change the initial heading and modified heading text as desired.
  const initialHeading = "Our First Test";
  const modifiedHeading = "Heading After Click";
  const [heading, setHeading] = useState(initialHeading);

  const handleClick = () => {
    setHeading(modifiedHeading);
  };

  return (
    <div>
      <button onClick={handleClick}>Click Me</button>
      <h1>{heading}</h1>
    </div>
  );
}

export default App;
