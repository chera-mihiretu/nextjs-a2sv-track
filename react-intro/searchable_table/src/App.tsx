import './App.css';
import FilterableProductTable from './components/FilterableProductTable';

function App() {
  return (
    <>
      <h1 className="font-sans text-[2.5rem] font-semibold text-[#4A90E2] text-center my-5">
        My Products
      </h1>
      <div className="App">
          <FilterableProductTable/>
      </div>
    </>
  );
}

export default App;
