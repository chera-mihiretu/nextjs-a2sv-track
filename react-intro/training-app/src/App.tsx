
import './App.css'
import YoutubeForm from './components/use_action_state/YoutubeForm'


function App() {

  return (
    <>
      <YoutubeForm/>
    </>
  )
}

export default App
