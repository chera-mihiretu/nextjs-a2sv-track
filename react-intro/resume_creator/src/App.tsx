import './App.css'
import EducationInfo from './components/EducationInfo';
import ExperienceInfo from './components/ExperienceInfo';
import PersonalInfo from './components/PersonalInfo';
import {useState} from 'react';
import ProjectInfo from './components/ProjectInfo';
import { EducationInformation, ExperienceInformation, PersonalInformation, ProjectInformation, UserData } from './components/UserData';





function App() {
  const [pageNumber, setPageNumber] = useState(0);
  const [userData, setUserData] = useState<UserData>({ 
    personal_information: {} as PersonalInformation, 
    education_information: {} as EducationInformation, 
    experience_information: {} as ExperienceInformation, 
    project_information: {} as ProjectInformation 
  });

  const nextPage = () => {

    setPageNumber((pageNumber + 1) % 4);
  }

  const prevPage = () => {
    if (pageNumber === 0 || pageNumber === 3) {
      return;
    }
    setPageNumber((pageNumber - 1) % 3);
  }

  function setProfessionalInfo(data: PersonalInformation) {
    const new_data : UserData = {...userData}
    new_data["personal_information"] = data;
    setUserData(new_data);
  }

  function setProjectInfo(data : ProjectInformation) {
    const new_data : UserData = {...userData}
    new_data["project_information"] = data;
    setUserData(new_data);
  }

  function setEducationInfo(data : EducationInformation) {
    const new_data : UserData = {...userData}
    new_data["education_information"] = data;
    setUserData(new_data);
  }

  function setExperianceInfo(data : ExperienceInformation) {
    const new_data : UserData = {...userData}
    new_data["experience_information"] = data;
    setUserData(new_data);
  }

  return (
    <>
      {pageNumber === 0 && (
        <div>
            <PersonalInfo nextPage={nextPage} setUserData={setProfessionalInfo} userData={userData}/>
        </div>
        )}
        {pageNumber === 1 && (
          <div>
            <EducationInfo nextPage={nextPage} prevPage={prevPage} setUserData={setEducationInfo} userData={userData}/>
          </div>
        )}
        {pageNumber === 2 && (
          <div>
            <ExperienceInfo nextPage={nextPage} prevPage={prevPage} setUserData={setExperianceInfo}userData={userData}/>
          </div>
        )}

        {pageNumber === 3 && (
          <div>
            <ProjectInfo nextPage={nextPage} prevPage={prevPage} setUserData={setProjectInfo} userData={userData}/>
          </div>
        )}
    </>
  )
}

export default App
