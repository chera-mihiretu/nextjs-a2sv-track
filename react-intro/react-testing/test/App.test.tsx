// App.test.jsx

import { vi, describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../src/App";
import React, { use } from "react";
import userEvent from "@testing-library/user-event";
import { Buttons } from "@testing-library/user-event/dist/cjs/system/pointer/buttons.js";
import CustomButton from "../src/components/CustomButton";

describe('App Componenets', () => {
    it('Should render magnificent monkeys', () => {
        const { container } = render(<App />);

        expect(container).toMatchSnapshot();
    });
    it('render radical rhinos after button click', async () => {

        const user = userEvent.setup();

        render(<App />);
        expect(screen.getByRole("heading").textContent).toMatch(/Our First Test/i)
        const button = screen.getByRole("button", {name : "Click Me"});

        await user.click(button);

        expect(screen.getByRole("heading").textContent).toBe("Heading After Click");

    });

    it("Should call the onClick when clicked", async () => {
        const onClick = vi.fn();
        const user = userEvent.setup()

        render(<CustomButton onClick={onClick} />);

        const button = screen.getByRole("button", {name : "Click me"});

        await user.click(button);

        expect(onClick).toHaveBeenCalled();
    });

    it("Should not call onclick when not clicked",  async () => {
        const onClick = vi.fn();
        const user = userEvent.setup()

        render(<CustomButton onClick={onClick} />);

        const button = screen.getByRole("button", {name : "Click me"});

 
        expect(onClick).not.toHaveBeenCalled();
    });
})
