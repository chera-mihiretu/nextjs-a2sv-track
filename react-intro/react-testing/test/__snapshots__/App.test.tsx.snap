// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Componenets > Should render magnificent monkeys 1`] = `
<div>
  <div>
    <button>
      Click Me
    </button>
    <h1>
      Our First Test
    </h1>
  </div>
</div>
`;
